src/main/scala/firrtl/transforms/ReplaceTruncatingArithmetic.scala

```diff
@@ -0,0 +1,53 @@
+package firrtl
+package transforms
+
+import firrtl.ir._
+import firrtl.Mappers._
+import firrtl.PrimOps._
+import firrtl.WrappedExpression._
+
+import scala.collection.mutable
+
+/** Replaces non-expanding arithmetic
+  *
+  * This replaces FIRRTL primops with ops that are not legal FIRRTL. They are useful for emission to
+  * languages that support non-expanding arithmetic (like Verilog)
+  */
+object ReplaceTruncatingArithmetic {
+
+  /** Mapping from references to the [[Expression]]s that drive them */
+  type Netlist = mutable.HashMap[WrappedExpression, Expression]
```

Comment:
Is there a reason this is WrappedExpression instead of Memoized?

Reply:
It's WrappedExpression so that WRefs that aren't fully filled out (like just filling out the name) will map to the same value, but it possibly should be `MemoizedHash[WrappedExpression]`